plugins/package-managers/node/src/main/kotlin/utils/NpmSupport.kt

```diff
@@ -153,15 +154,22 @@ internal fun parseNpmVcsInfo(node: JsonNode): VcsInfo {
         revision = head
     )
 
-    val type = repository["type"].textValueOrEmpty()
-    val url = repository.textValue() ?: repository["url"].textValueOrEmpty()
+    if (repository.isTextual) {
+        val url = expandNpmShortcutUrl(repository.textValue())
+        val vcsInfo = VcsHost.parseUrl(url)
+        return vcsInfo.copy(revision = head.takeIf { it.isSha1() } ?: vcsInfo.revision.takeIf { it.isSha1() }.orEmpty())
+    }
+
+    val type = VcsType.forName(repository["type"].textValueOrEmpty()).takeUnless { it == VcsType.UNKNOWN }
```

Comment:
At this point, we know that `repository` is an object node with mandatory `type` and `url` nodes, so we could use just `textValue()` instead of `textValueOrEmpty()` here any below.

Reply:
Maybe move the `.takeUnless { it == VcsType.UNKNOWN }` part to line 169 to highlight why the `?:` fallback is needed there?